File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/socket/to/helper/IntMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class IntMapHelper {

	public static int getByteArrayLength(Map<String, Integer> intMap){
		int blength = 0;
		Iterator<String> keyIt = intMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			blength += TransferUtil.getLengthOfInt();
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, Integer> intMap, TransferOutputStream touts) {
		Iterator<String> keyIt = intMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_INT);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			int value = intMap.get(key);
			touts.writeInt(value);
		}
	}
}

File: history/0.1.2/tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/IntMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class IntMapHelper {
        public static int getByteArrayLength(Dictionary<String, Int32> intMap) {
            int blength = 0;
            foreach (KeyValuePair<String, Int32> item in intMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                blength += TransferUtil.getLengthOfInt();
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, Int32> intMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, Int32> item in intMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_INT);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                int value = item.Value;
                touts.writeInt(value);
            }
        }
    }
}